firebase_admin/project_management.py

```diff
@@ -0,0 +1,357 @@
+        self._project_id = _check_is_nonempty_string(project_id, 'project_id')
+
+    @property
+    def name(self):
+        """The fully qualified resource name of this Android App."""
+        return self._name
+
+    @property
+    def app_id(self):
+        """The globally unique, Firebase-assigned identifier of this Android App.
+
+        This ID is unique even across Apps of different platforms, such as iOS Apps.
+        """
+        return self._app_id
+
+    @property
+    def display_name(self):
+        """The user-assigned display name of this Android App."""
+        return self._display_name
+
+    @property
+    def project_id(self):
+        """The permanent, globally unique, user-assigned ID of the parent Firebase Project."""
+        return self._project_id
+
+
+class AndroidAppMetadata(AppMetadata):
+    """Android-specific information about an Android Firebase App."""
+
+    def __init__(self, name, app_id, display_name, project_id, package_name):
+        super(AndroidAppMetadata, self).__init__(name, app_id, display_name, project_id)
+        self._package_name = _check_is_nonempty_string(package_name, 'package_name')
+
+    @property
+    def package_name(self):
+        """The canonical package name of this Android App as it would appear in the Play Store."""
+        return self._package_name
+
+
+class _ProjectManagementService(object):
+    """Provides methods for interacting with the Firebase Project Management Service."""
+
+    BASE_URL = 'https://firebase.googleapis.com'
+    MAXIMUM_LIST_APPS_PAGE_SIZE = 1
+    ERROR_CODES = {
+        401: 'Request not authorized.',
+        403: 'Client does not have sufficient privileges.',
+        404: 'Failed to find the resource.',
+        409: 'The resource already exists.',
+        429: 'Request throttled out by the backend server.',
+        500: 'Internal server error.',
+        503: 'Backend servers are over capacity. Try again later.'
+    }
+
+    def __init__(self, app):
+        project_id = app.project_id
+        if not project_id:
+            raise ValueError(
+                'Project ID is required to access the Firebase Project Management Service. Either '
+                'set the projectId option, or use service account credentials. Alternatively, set '
+                'the GOOGLE_CLOUD_PROJECT environment variable.')
+        self._project_id = project_id
+        self._client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_ProjectManagementService.BASE_URL)
+        self._timeout = app.options.get('httpTimeout')
+
+    def get_android_app_metadata(self, app_id):
+        """Retrieves detailed information about an Android App."""
+        _check_is_nonempty_string(app_id, 'app_id')
+        path = '/v1beta1/projects/-/androidApps/{0}'.format(app_id)
+        response = self._make_request('get', path, app_id, 'App ID')
+        return AndroidAppMetadata(
```

Comment:
In Node.js or Java I would have said provide a constructor that takes the response as the argument. But methods with many keyword args are idiomatic in Python, so this is ok.